### PR TITLE
Fix nil pointer when processing invalid waypoint backendrefs

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
Service may be nil, so  refactor to prevent panics

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**
- [X] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### CODE REVIEW

File: pilot/pkg/networking/core/listener_waypoint.go [hunk 1]
```diff
@@ -701,9 +701,13 @@ func (lb *ListenerBuilder) waypointRouteDestination(
 		action.Timeout = in.Timeout
 	}
 	// Use deprecated value for now as the replacement MaxStreamDuration has some regressions.
+	// TODO: check and see if the replacement has been fixed.
 	// nolint: staticcheck
 	action.MaxGrpcTimeout = action.Timeout
 
+	// return 500 for invalid backends
+	// https://github.com/kubernetes-sigs/gateway-api/blob/cea484e38e078a2c1997d8c7a62f410a1540f519/apis/v1beta1/httproute_types.go#L204
+	action.ClusterNotFoundResponseCode = route.RouteAction_INTERNAL_SERVER_ERROR
```
Comment (howardjohn) [inline pilot/pkg/networking/core/listener_waypoint.go:710]: Should be only if gatewaySemantics?

Comment (keithmattix) [reply pilot/pkg/networking/core/listener_waypoint.go:710]: Don't waypoints always have gateway semantics?

Comment (howardjohn) [reply pilot/pkg/networking/core/listener_waypoint.go:710]: Hmm, you can use VirtualService. Or have no explicit routes at all

Comment (keithmattix) [reply pilot/pkg/networking/core/listener_waypoint.go:710]: Fair point - will guard with an if

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
I am no sure this is needed for 1.23 btw

**Comment by keithmattix** [comment]
@howardjohn yep when I saw the failed cherrypicks I looked at the code and concluded that it was a 1.24 only regression

**Pull request merged**